src/createTimestampRecord.ts

```diff
@@ -1,57 +1,54 @@
 import logger from './util/logger';
-import _ from 'lodash';
-import { LegacyTechRecord, SingleTechRecord } from './Interfaces/ILegacyTechRecord';
-import { NewKeyStructure } from './Interfaces/INewTechRecord';
+import { LegacyVehicleRecord, LegacyTechnicalRecord } from './interfaces/LegacyVehicleRecord';
+import { NewVehicleRecord } from './interfaces/NewVehicleRecord';
+import { PrimitiveTypes } from './interfaces/PrimitiveTypes';
 
-const isValidValue = (a: unknown) => {
-  return a !== null && a !== undefined && (_.isString(a) || _.isNumber(a) || _.isBoolean(a));
+const isObject = (a: unknown): a is object => {
+  const type = typeof a;
+  return type !== 'function' && type === 'object';
```

Comment:
```suggestion
  return type !== 'function' && type === 'object' && a !== null;
```
I know in this instance it would be caught before by the `isPrimitive`, just a suggestion

Reply:
It makes sense for completeness, but I don't particularly want to add an additional condition to something I'm trying to optimise if the condition isn't reachable in this context.